modules/gateway/persist.go

```diff
@@ -14,12 +14,25 @@ func (g *Gateway) save() error {
 	for node := range g.nodes {
 		nodes = append(nodes, node)
 	}
-	return encoding.WriteFile(filepath.Join(g.saveDir, "nodes.dat"), nodes)
+	data, err := json.MarshalIndent(nodes, "", "\t")
+	if err != nil {
+		return err
+	}
+	file, err := os.Create(filepath.Join(g.saveDir, "nodes.json"))
+	if err != nil {
+		return err
+	}
+	_, err = file.Write(data)
+	return err
 }
```

Comment:
This change is compatibility breaking. We should have some way to document all the places we break compatibility. I think a list could be useful when debugging user problems.

Maybe a changelog?